src/java/voldemort/server/protocol/admin/AdminServiceRequestHandler.java

```diff
@@ -1162,6 +1172,9 @@ public void operate() {
                         } catch(Exception e) {
                             throw new VoldemortException("Exception in Fetcher = " + e.getMessage(),
                                                          e);
+                        } finally{
+                            if (store.isCurrentlyFetching())
```

Comment:
Very good thinking to put this in the finally clause. However, I think you should remove the other ```store.setCurrentlyFetching(false)``` from line 1148, just above, since it makes it possible that another thread would set the flag to true between 1148 and 1177 and then the flag would be clobbered back to false in the finally clause.